gordon/schema/validate.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2017 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from gordon import exceptions
+
+import json
+import os
+
+import jsonschema
+
+DEFINITIONS_DIRECTORY = os.path.join(os.path.dirname(__file__), 'definitions')
+
+
+class JsonValidator(object):
+
+    def __init__(self):
+        """Initialize known schemas from files into reference dict."""
+        self.schemas = {}
+        for file in os.listdir(DEFINITIONS_DIRECTORY):
```

Comment:
Have you considered using [pathlib.Path](https://docs.python.org/3/library/pathlib.html) here? It provides a higher level way of working with paths than using plain strings, makes for slightly easier usage and testing. For example, this functionality could look like:

```python
from pathlib import Path
DEFINITIONS_DIRECTORY = Path('.', 'definitions').absolute()
# ...
for definition in DEFINITIONS_DIRECTORY.glob('*.json'):
    with open(definition, 'r') as f:
        self.schemas[definition.stem] = json.loads(f.read())
```

Reply:
Nope, I'd never even seen it before.  I don't immediately see the benefits, but I've got nothing against trying something new

Reply:
This is a bit late but... [puts on salesman hat]
From messing around with it, I've found that it ups the understandability of the the code. Instead of passing around strings everywhere and calling `os` or string functions to get things like the parent directory or checking for existence, you get nice objects that you can ask things. Eg.

```python
path = '/etc/supervisor'
config_path = os.path.join(path, 'config.conf')
os.exists(config_path)
parent_dir = config_path.split('/')[-1]
# vs
path = Path('/etc/supervisor')
config_path = path / 'config.conf'
config_path.exists()
parent_dir = config_path.parent
```

Found that this makes tests that work with paths simpler, too.
[/salesman hat] :)